pkgs/tools/filesystems/bees/bees-service-wrapper

```diff
@@ -0,0 +1,196 @@
+# shouldn't need to persist something that will differ between hosts built from
+# the same configuration template.
+
+# Thus, for using bees from NixOS, we have our own wrapper, which supports not
+# just UUID but any specification permitted by findmnt
+
+[[ $bees_debug ]] && { PS4=':${BASH_SOURCE##*/}:$LINENO+'; set -x; }
+
+usage() {
+  cat >&2 <<EOF
+Usage: ${BASH_SOURCE##*/} run|cleanup config-name|fs-spec [idxSizeMB=...] [verbosity=...] [workDir=...] [-- options...]
+
+  fs-spec should be in a format recognized by findmnt.
+EOF
+  exit 1
+}
+
+die() { echo "$*" >&2; exit 1; }
+
+# Alternate names for configuration values
+declare -A configNames=(
+  # from original bees wrapper
+  [BEESHOME]=bees_home
+  [BEESSTATUS]=bees_status
+  [MNT_DIR]=bees_mnt_dir
+  [UUID]=bees_uuid
+  [WORK_DIR]=bees_run_dir
+  # from command-line usage
+  [idxSizeMB]=bees_db_size_mb
+  [verbosity]=bees_verbosity
+  [workDir]=bees_work_dir
+)
+
+# legacy bees config files can be arbitrary shell scripts, so we need to actually evaluate them
+readConfigFileIfExists() {
+  [[ -s $1 ]] || return 1
+  source "$1"
+  for oldName in "${!configNames[@]}"; do : "$oldName"
+    if [[ ${!oldName} ]]; then
+      newName=${configNames[$oldName]}
+      printf -v "$newName" %s "${!oldName}"
+      unset "$oldName"
+    fi
+  done
+}
+
+uuid_re='^[[:xdigit:]]{8}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{12}$'
+
+# Shared code for setting configuration used by other operations.
+#
+# Reads from global associative array "options" containing options passed in as
+# key=value pairs on the command line, looks for config-file overrides, and
+# sets individual global variables.
+_setup() {
+  declare fstype fs_spec=$1; shift
+  declare -g bees_{blockdev,uuid,verbosity} bees_{work,run,mnt}_dir bees_db_size{,_mb}
+  for key in "${!options[@]}"; do
+    # Rename options as appropriate
+    if [[ ${configNames[$key]} && ${configNames[$key]} != "$key" ]]; then
+      options[${configNames[$key]}]=${options[$key]}
+      unset "options[$key]"
+    fi
+  done
+
+  # Look for file-based configuration, additional to honoring configuration on the command line
+  bees_config_dir="${bees_config_dir:-/etc/bees}"
+  if [[ $fs_spec =~ $uuid_re ]]; then
+    bees_uuid=$fs_spec
+    # If our spec looks like a bare UUID, and no config file exists in the new
+    # format, fall back to legacy config file search mechanism (grep; ewww).
+    if ! readConfigFileIfExists "$bees_config_dir/UUID=$fs_spec.conf"; then
+      # Legacy approach to finding a config file: Grep for a *.conf file
+      # containing the UUID within its text. Permitting spaces around the "="
+      # appears to be a bug, but is retained for compatibility with the
+      # original upstream script.
+      allConfFiles=( "$bees_config_dir"/*.conf )
+      if (( ${#allConfFiles[@]} )); then
+        readarray -d '' -t matchingConfFiles < <(grep -E -l -Z "^[^#]*UUID[[:space:]]*=[[:space:]]*" "${allConfFiles[@]}"
```

Comment:
Where is the closing paren for `<(` ?

Reply:
Argh! Lost it in the rewrite; that's what I get for considering something too trivial a change to test. :(